src/Fixer/StringNotation/SingleQuoteFixer.php

```diff
@@ -54,17 +54,46 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
             }
 
             $content = $token->getContent();
+            $firstCharacter = $content[0];
 
             if (
-                '"' === $content[0] &&
-                false === strpos($content, "'") &&
-                // regex: odd number of backslashes, not followed by double quote or dollar
-                !Preg::match('/(?<!\\\\)(?:\\\\{2})*\\\\(?!["$\\\\])/', $content)
+                '"' === $firstCharacter &&
+                $this->isCompatibleWithSingleQuotes($content)
             ) {
                 $content = substr($content, 1, -1);
-                $content = str_replace(array('\\"', '\\$'), array('"', '$'), $content);
+                $content = $this->addSlashesToCharacters($content);
                 $tokens[$index] = new Token(array(T_CONSTANT_ENCAPSED_STRING, '\''.$content.'\''));
             }
+
+            if (
```

Comment:
The body of this `if` and the one before it are pretty much the same except handling of first character. Removing the `b` flag from `$content` if present, then keeping existing process and adding the flag back would allow you to do less changes.

Reply:
Good point, fixed!